SIGSEGV on package repo install

## Bug Report

Running `tanzu package repository install --default`, getting the error:

```
❯ tanzu package repository install --default
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4eade66]

goroutine 1 [running]:
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.mutate(0x0, 0x0, 0x0, 0x5583a3d, 0x15, 0x578e5b0, 0xc000442b40, 0x0, 0x0)
	/Users/smcginnis/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.14/pkg/controller/controllerutil/controllerutil.go:228 +0x26
sigs.k8s.io/controller-runtime/pkg/controller/controllerutil.CreateOrUpdate(0x57b7bb8, 0xc0000460d0, 0x57c7d38, 0xc00010f3b0, 0x578e5b0, 0xc000442b40, 0x0, 0x40a1624, 0xc00013c3d8, 0xc0006dfb68, ...)
	/Users/smcginnis/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.14/pkg/controller/controllerutil/controllerutil.go:202 +0x1cf
github.com/vmware-tanzu/tce/cli/pkg/common/kapp.(*Kapp).InstallRepository(0xc00013c3c0, 0x5583a3d, 0x15, 0x55a496d, 0x29, 0x0, 0x0)
	/Users/smcginnis/src/vmware-tanzu/tce/cli/pkg/common/kapp/repository.go:92 +0x4b2
github.com/vmware-tanzu/tce/cli/pkg/common/kapp.(*Kapp).InstallDefaultRepository(...)
	/Users/smcginnis/src/vmware-tanzu/tce/cli/pkg/common/kapp/repository.go:63
github.com/vmware-tanzu/tce/cli/pkg/addon.installRepository(0x62227a0, 0xc000126a20, 0x0, 0x1, 0x0, 0x0)
	/Users/smcginnis/src/vmware-tanzu/tce/cli/pkg/addon/repository.go:54 +0x139
github.com/spf13/cobra.(*Command).execute(0x62227a0, 0xc000126a10, 0x1, 0x1, 0x62227a0, 0xc000126a10)
	/Users/smcginnis/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004c4000, 0xc0006dff50, 0x5, 0x5)
	/Users/smcginnis/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/smcginnis/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/vmware-tanzu-private/core/pkg/v1/cli/command/plugin.(*Plugin).Execute(...)
	/Users/smcginnis/go/pkg/mod/github.com/vmware-tanzu-private/core@v1.3.0/pkg/v1/cli/command/plugin/plugin.go:49
main.main()
	/Users/smcginnis/src/vmware-tanzu/tce/cli/cmd/plugin/addon/main.go:39 +0x1b8
Error: exit status 2
```

This appears to be due to an update to a controller API method. It expects a function to be passed in to mutate objects, but we are passing in nil.

## Expected Behavior

Repo installed successfully.

## Steps to Reproduce the Bug

Run `tanzu package repository install --default`

## Environment Details

* Build version (`tanzu version`):  current dev build
* Operating System (client): darwin

--- a/cli/pkg/common/kapp/kapp.go
+++ b/cli/pkg/common/kapp/kapp.go
@@ -231,7 +231,7 @@ func (k *Kapp) installServiceAccount(kClient client.Client, input *AppCrdInput) 
 	klog.V(6).Infof("serviceAccount.Name = %s", serviceAccount.ObjectMeta.Name)
 	klog.V(6).Infof("sa.Namespace = %s", serviceAccount.ObjectMeta.Namespace)
 
-	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, serviceAccount, nil)
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, serviceAccount, mutate)
 	if err != nil {
 		klog.Errorf("Error creating or patching addon service account. Err: %v", err)
 		return err
@@ -265,7 +265,7 @@ func (k *Kapp) installRoleBinding(kClient client.Client, input *AppCrdInput) err
 	klog.V(6).Infof("roleBinding.Subjects.Name = %s", roleBinding.Subjects[0].Name)
 	klog.V(6).Infof("roleBinding.Subjects.Namespace = %s", roleBinding.Subjects[0].Namespace)
 
-	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, roleBinding, nil)
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, roleBinding, mutate)
 	if err != nil {
 		klog.Errorf("Error creating or patching addon role binding. Err: %v", err)
 		return err
@@ -348,7 +348,7 @@ func (k *Kapp) installInstalledPackage(kClient client.Client, input *AppCrdInput
 	}
 
 	klog.V(6).Infof("Deploying installed package: %s", ip)
-	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, ip, nil)
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, ip, mutate)
 	if err != nil {
 		return err
 	}
@@ -486,7 +486,7 @@ func (k *Kapp) installConfigSecret(kClient client.Client, input *AppCrdInput) (*
 		},
 	}
 
-	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, config, nil)
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), kClient, config, mutate)
 	if err != nil {
 		return nil, err
 	}

--- a/cli/pkg/common/kapp/repository.go
+++ b/cli/pkg/common/kapp/repository.go
@@ -89,7 +89,7 @@ func (k *Kapp) InstallRepository(name, url string) error {
 		},
 	}
 
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), client, repo, nil)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), client, repo, mutate)
 	if err != nil {
 		klog.Errorf("Create failed. Err: %v", err)
 		return err
@@ -116,7 +116,7 @@ func (k *Kapp) InstallRepositoryFromFile(filename string) error {
 		return err
 	}
 
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), client, repo, nil)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), client, repo, mutate)
 	if err != nil {
 		klog.Errorf("Create failed. Err: %v", err)
 		return err

--- a/cli/pkg/common/kapp/types.go
+++ b/cli/pkg/common/kapp/types.go
@@ -52,3 +52,13 @@ type Kapp struct {
 	// localWorkingDirectory is where all modification based on user overrides is defined
 	localWorkingDirectory string
 }
+
+// mutate is used to pass in to controllerutil calls to mutate an object into
+// its desired state. For now it is just a noop since we don't need to do
+// anything, but some calls expect a function to be passed in and will panic if
+// nil. If we don't find a need for mutating, this can be cleaned up once
+// https://github.com/kubernetes-sigs/controller-runtime/issues/1476 is
+// resolved.
+func mutate() error {
+	return nil
+}

